python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED.py

```diff
@@ -0,0 +1,526 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+ Rule Name:
+   IAM_POLICY_REQUIRED
+ Description:
+   To check IAM users and roles have a given policy attached directly or through a group.
+ Trigger:
+   Configuration Change on AWS::IAM::User/AWS::IAM::Role
+ Reports on:
+   AWS::IAM::User,AWS::IAM::Role
+ Rule Parameters:
+   | ---------------------- | --------- | -------------------------------------------------------- |
+   | Parameter Name         | Type      | Description                                              |
+   | ---------------------- | --------- | -------------------------------------------------------- |
+   | policyArns             | Required  | ARNs of the policies which should be attached to all     |
+   |                        |           | users and roles.                                         |
+   | ---------------------- | --------- | -------------------------------------------------------- |
+   | exceptionList          | Optional  | Represents the IAM users and roles which are exempted    |
+   |                        |           | from the IAM Config rule. The valid users in this list   |
+   |                        |           | will be compliant by default.                            |
+   |                        |           | List of all the User Names separated by a comma          |
+   | ---------------------- | --------- | -------------------------------------------------------- |
+ Feature:
+   As: a Security Officer
+   I want: To ensure that IAM roles and users have mandatory policies attached
+   In order to: enforce mandatory permissions for IAM entities
+ Scenarios:
+  Scenario: 1
+     Given: No IAM Users or Roles exist
+      When: Evaluation occurs
+      Then: Return "NOT_APPLICABLE"
+   Scenario: 2
+     Given: exceptionList is configured
+       And: A user listed in the exceptionList is not an alphanumerical string
+      When: Evaluation occurs
+      Then: Return an error
+   Scenario: 3
+     Given: exceptionList is configured
+       And: A role listed in the exceptionList is not an alphanumerical string

```

Comment:
For scenario having just one term which change, you can use the scenario outline format: https://stackoverflow.com/questions/9093872/gherkin-or-syntax-to-reduce-repetition-with-bdd

```
   Scenario: 2
     Given: exceptionList is configured
       And: A <entity> listed in the exceptionList is not an alphanumerical string
      Then: Return an error

   Examples:
  | entity     |
  | IAM User   |
  | IAM Role   |
```